internal/util/util.go

```diff
@@ -88,8 +89,38 @@ func ShootFromCAPIResources(
 	}
 }
 
+var (
+	// AnnotationAllowList defines the list of annotations that are allowed to be synced between GardenerShootControlPlane and Shoot.
+	// This is used to ensure that only specific annotations are propagated to the Shoot resource.
+	AnnotationAllowList = []string{
+		gardenerv1beta1constants.GardenerOperation,
+	}
+)
+
+func syncAnnotations(source, target map[string]string, allowedKeys []string) map[string]string {
+	// Remove annotations from target that are not in the source and are in the allowed list
+	for _, key := range allowedKeys {
+		if _, exists := target[key]; exists {
+			if _, existsInSource := source[key]; !existsInSource {
+				delete(target, key)
+			}
+		}
+	}
+
+	// Add or update annotations from source to target if they are in the allowed list
+	for _, key := range allowedKeys {
+		if value, exists := source[key]; exists {
+			target[key] = value
+		}
+	}
```

Comment:
Humble suggestion for improved readability:

```suggestion
	for _, key := range allowedKeys {
		// Add or update annotations from source to target if they are in the allowed list		
		if sourceValue, existsInSource := source[key]; existsInSource {
			target[key] = sourceValue
			continue
		}

		// Remove annotations from target that are not in the source and are in the allowed list
		_, existsInTarget := target[key]
		if existsInTarget {
			delete(target, key)
		}
	}
```